cmd/main.go

```diff
@@ -353,6 +379,67 @@ func RunRootCommand(ctx context.Context, opt Options, args []string) error {
 	}
 	defer llmClient.Close()
 
+	// Initialize session management
+	var currentSession *sessions.Session
+	var sessionManager *sessions.SessionManager
+
+	// TODO: Remove this when session persistence is default
+	if opt.NewSession || opt.ResumeSession != "" {
+		sessionManager, err = sessions.NewSessionManager()
+		if err != nil {
+			return fmt.Errorf("failed to create session manager: %w", err)
+		}
+
+		// Handle session creation or loading
+		if opt.NewSession {
+			// Create a new session
+			meta := sessions.Metadata{
+				ProviderID: opt.ProviderID,
+				ModelID:    opt.ModelID,
+			}
+			currentSession, err = sessionManager.NewSession(meta)
+			if err != nil {
+				return fmt.Errorf("failed to create a new session: %w", err)
+			}
+			klog.Infof("Created new session: %s\n", currentSession.ID)
+		} else {
+			// Load existing session
+			var sessionID string
+			if opt.ResumeSession == "" || opt.ResumeSession == "latest" {
+				// Get the latest session
+				currentSession, err = sessionManager.GetLatestSession()
+				if err != nil {
+					return fmt.Errorf("failed to get latest session: %w", err)
+				}
+				if currentSession == nil {
+					// No sessions exist, create a new one
+					meta := sessions.Metadata{
+						ProviderID: opt.ProviderID,
+						ModelID:    opt.ModelID,
+					}
+					currentSession, err = sessionManager.NewSession(meta)
+					if err != nil {
+						return fmt.Errorf("failed to create new session: %w", err)
+					}
+					klog.Infof("Created new session: %s\n", currentSession.ID)
+				}
+			} else {
+				sessionID = opt.ResumeSession
+				currentSession, err = sessionManager.FindSessionByID(sessionID)
+				if err != nil {
+					return fmt.Errorf("session %s not found: %w", sessionID, err)
+				}
+			}
+
+			if currentSession != nil {
+				// Update last accessed time
+				if err := currentSession.UpdateLastAccessed(); err != nil {
+					klog.Warningf("Failed to update session last accessed time: %v", err)
+				}
+			}
+		}
+	}
```

Comment:
I think it is safe to assume if user has not specified `--new-session` or `--resume-session`, they don't exist session to be persisted and we can safely assign `currentSession` to in-memory chat store. This will eliminate the `nil` checks for agent.ChatMessageStore later in the code.